design/one-pager-stacks-security-isolation.md

```diff
@@ -0,0 +1,187 @@
+
+> CustomResourceDefinitions themselves are non-namespaced and are available to all namespaces
+
+Therefore, any time a CRD is created, it will also be at the cluster scope, even if instances of the CRD are scoped to a namespace.
+This has caused confusion in the past, so it's important to clear that up here.
+
+## Platform Stacks
+
+Platform Stacks can be thought of as a higher level of privilege than other types of stacks, similar to how kernel modules work in Linux.
+The controllers contained in a platform stack will usually reconcile resources that are external to the cluster, for example cloud provider managed services.
+These controllers will also likely be written as full fledged controller-runtime based implementations, but that is not a requirement.
+
+Let's look at a specific example of the GCP platform stack to understand the security and isolation model more thoroughly.
+When the Stack Manager installs the GCP stack, a few things will happen:
+
+* All CRDs related to GCP infrastructure and services will be installed to the cluster, such as `CloudSQLInstance`, `CloudSQLInstanceClass`, `GKECluster`, `GKEClusterClass`, `Network`, `Subnetwork`, etc.
+* A `ClusterRole` will be created at the cluster level that includes all of the RBAC rules for API groups, resources, and verbs that were specified in the `rbac.yaml` file from the stack.
+Because this is a `ClusterRole`, bindings to this role can grant the given permissions to cluster-scoped resources and to namespaced resources across all namespaces.
+* A `ServiceAccount` is created in the namespace that the stack's controller will be running.  This will serve as the runtime identify of the controller.
+* A `ClusterRoleBinding` is created at the cluster level to bind the `ClusterRole` and the namespaced `ServiceAccount`
+* A `Deployment` is created in the same namespace as the stack and it is expected to launch and manage the pod that hosts the GCP controllers.
+
+It is worth reiterating the scope of permissions that this platform stack has received:
+
+> Because the stack is given a `ClusterRole`, it can be granted permissions to cluster-scoped resources and to namespaced resources across all namespaces.
+
+### Architecture Diagram - Platform Stack
+
+The diagram below summarizes the artifacts in the control plane after the GCP stack has been installed and a couple example `MySQLClaims` have been created.
+
+![Diagram of a Platform Stack Installed](./images/platform-stack-installed.png)
+
+### Environments
+
+While platform stacks are installed at the cluster level (there is only one instance of each platform stack), they can still support the concept of multiple "environments", each represented by a namespace.
+For example, an admin can create a namespace to model their production "environment" and another namespace to model their development "environment".
+All resources within each of those environments can share a common set of provider credentials.
```

Comment:
Would we want these credentials (i.e. the `Provider` instance) to be cluster-scoped, or do you envision them just living in the `gcp` (or other) namespace and being referenced by resource classes by namespaced `ObjectReference`?